plugin.py

```diff
@@ -1,101 +1,44 @@
+    server_directory = 'vscode-json-languageserver'
+    server_binary_path = os.path.join(
+        server_directory, 'node_modules', 'vscode-json-languageserver', 'bin', 'vscode-json-languageserver'
+    )
 
-class LspJSONPlugin(LanguageHandler):
     def __init__(self) -> None:
+        super().__init__()
         self._default_schemas = []  # type: List[Dict]
 
-    @property
-    def name(self) -> str:
-        return PACKAGE_NAME.lower()
-
-    @property
-    def config(self) -> ClientConfig:
-        # Calling setup() also here as this might run before `plugin_loaded`.
-        # Will be a no-op if already ran.
-        # See https://github.com/sublimelsp/LSP/issues/899
-        server.setup()
-
-        configuration = self.migrate_and_read_configuration()
-
-        default_configuration = {
-            'enabled': True,
-            'command': ['node', server.binary_path, '--stdio'],
-        }
-
-        default_configuration.update(configuration)
-
+    def on_client_configuration_ready(self, configuration: Dict):
         if not self._default_schemas:
             schemas = ['schemas_extra.json', 'schemas.json']
             for schema in schemas:
-                path = 'Packages/{}/{}'.format(PACKAGE_NAME, schema)
+                path = 'Packages/{}/{}'.format(self.package_name, schema)
                 self._default_schemas.extend(json.loads(ResourcePath(path).read_text()))
 
-        default_configuration['settings'].setdefault('json', {})['schemas'] = self._default_schemas
-
-        return read_client_config(self.name, default_configuration)
-
-    def migrate_and_read_configuration(self) -> dict:
-        settings = {}
-        loaded_settings = sublime.load_settings(SETTINGS_FILENAME)
-
-        if loaded_settings:
-            if loaded_settings.has('client'):
-                client = loaded_settings.get('client')
-                loaded_settings.erase('client')
-                # Migrate old keys
-                for key in client:
-                    loaded_settings.set(key, client[key])
-                sublime.save_settings(SETTINGS_FILENAME)
-
-            # Read configuration keys
-            for key in ['languages', 'initializationOptions', 'settings']:
-                settings[key] = loaded_settings.get(key)
-
-        return settings
-
-    def on_start(self, window) -> bool:
-        if not is_node_installed():
-            sublime.status_message('Please install Node.js for the JSON server to work.')
-            return False
-        return server.ready
+        configuration['settings'].setdefault('json', {})['schemas'] = self._default_schemas
 
-    def on_initialized(self, client) -> None:
-        client.on_request(
+    def on_ready(self, api) -> None:
+        api.on_request(
             'vscode/content',
-            lambda params, request_id: self.handle_vscode_content(client, params, request_id))
+            lambda params, request_id: self.handle_vscode_content(params, request_id))
```

Comment:
```
api.on_request(
            'vscode/content',
            self.handle_vscode_content)
```
Shouldn't this also also work? :)

Reply:
Yes. :)
And that was also confusing as argument names were wrong :)